packages/test/test-end-to-end-tests/src/test/t9sclient.spec.ts

```diff
@@ -0,0 +1,99 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { strict as assert } from "assert";
+import { describeNoCompat } from "@fluid-internal/test-version-utils";
+import { TinyliciousClient } from "@fluidframework/tinylicious-client";
+import { DefaultSummaryConfiguration } from "@fluidframework/container-runtime";
+import { IContainer } from "@fluidframework/container-definitions";
+import { SharedMap } from "@fluidframework/map";
+import { IClient } from "@fluidframework/protocol-definitions";
+import { ITestObjectProvider, waitForContainerConnection } from "@fluidframework/test-utils";
+
+describeNoCompat("Tinylicious client", (getTestObjectProvider) => {
+	let provider: ITestObjectProvider;
+	beforeEach(() => {
+		provider = getTestObjectProvider();
+	});
+
+	it("interactive client doesn't change its client details when summarizer starts", async function () {
+		// This test only applies to tinylicious
+		if (provider.driver.type !== "tinylicious") {
+			this.skip();
+		}
+		// This test validates behavior that only occurs when the summarizer is created with the same loader that created
+		// the container.
+		// Manually creating a summarizer (Summarizer.create()) requires passing in a loader but Tinylicious client does not
+		// keep a reference to the loader it uses to create containers.
+		// Instead we can tweak the summarizer heuristics for the container runtime so the summarizer spawns quickly on its
+		// own, and the test doesn't have to wait for an indeterminate amoun of time (up to several second) before it
+		// can validate things after the summarizer exists.
+
+		const tinyliciousClient = new TinyliciousClient({
```

Comment:
Instead of testing here, would be better to test at the `Container` layer (this repros for any customer who includes `options.client` regardless of whether they're using TinyliciousClient, AzureClient, or none of the above).

Reply:
i also don't know if we have any tests for client options override, so a general test that runs against all servers would give the best coverage.

Reply:
I'll look into that approach.